### PR TITLE
Improved library loading a bit

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

File: python-package/xgboost/core.py [hunk 1]
```diff
@@ -114,10 +114,16 @@ def _get_log_callback_func():
...
-    lib_path = find_lib_path()
-    if len(lib_path) == 0:
+    lib_paths = find_lib_path()
+    if len(lib_paths) == 0:
         return None
-    lib = ctypes.cdll.LoadLibrary(lib_path[0])
+    pathBackup = os.environ['PATH']
+    for lib_path in lib_paths:
+      try:
+        os.environ['PATH'] = pathBackup + os.pathsep + os.path.dirname(lib_path) # needed when the lib is compiled with non-system-available dependencies
```
Comment (hcho3) [inline python-package/xgboost/core.py]: "needed when the lib is compiled with non-system-available dependencies"
Which dependency was causing issue?

**Comment by KOLANICH** [comment]
libc from MinGW. I have manually added its dlls to the dir of xgboost.dll since putting them into path would break the apps dependent on other version of libc.

**Comment by hcho3** [comment]
@KOLANICH Can you fix the indentation of core.py? Existing Python code uses 4 spaces for indent.

File: python-package/xgboost/core.py [hunk 2]
```diff
@@ -114,10 +114,16 @@ def _get_log_callback_func():
...
-    lib_path = find_lib_path()
-    if len(lib_path) == 0:
+    lib_paths = find_lib_path()
+    if len(lib_paths) == 0:
         return None
-    lib = ctypes.cdll.LoadLibrary(lib_path[0])
+    pathBackup = os.environ['PATH']
+    for lib_path in lib_paths:
+        try:
+            os.environ['PATH'] = pathBackup + os.pathsep + os.path.dirname(lib_path) # needed when the lib is compiled with non-system-available dependencies
```
Comment (hcho3) [inline python-package/xgboost/core.py]: Can we put the comment in another line to pass the pylint?

File: python-package/xgboost/core.py [hunk 3]
```diff
@@ -114,10 +114,16 @@ def _get_log_callback_func():
...
+            lib = ctypes.cdll.LoadLibrary(lib_path)
+        except:
```
Comment (hcho3) [inline python-package/xgboost/core.py]: Let's specify the type of exception we'd like to catch here. (If the type is not known, just use `except Exception:`.

File: python-package/xgboost/core.py (hunk 2 above)
Comment (KOLANICH) [reply python-package/xgboost/core.py]: we can

File: python-package/xgboost/core.py (hunk 3 above)
Comment (KOLANICH) [reply python-package/xgboost/core.py]: OK

**Pull request merged**